WooCommerce/WooCommerceTests/ViewRelated/Orders/Order Creation/Synchronizer/RemoteOrderSynchronizerTests.swift

```diff
@@ -335,6 +528,80 @@ final class RemoteOrderSynchronizerTests: XCTestCase {
         }
     }
 
+    func test_sending_existing_productID_input_sends_order_with_totals() {
+        // Given
+        let stores = MockStoresManager(sessionManager: .testingInstance)
+        let synchronizer = RemoteOrderSynchronizer(siteID: sampleSiteID, flow: .creation, stores: stores)
+
+        // When
+        let submittedItems: [OrderItem] = waitFor { promise in
+            stores.whenReceivingAction(ofType: OrderAction.self) { action in
+                switch action {
+                case .createOrder(_, let order, let completion):
+                    completion(.success(order.copy(orderID: self.sampleOrderID)))
+                case .updateOrder(_, let order, _, _):
+                    promise(order.items)
+                default:
+                    XCTFail("Unexpected Action received: \(action)")
+                }
+            }
+
+            let initialInput = OrderSyncProductInput(id: self.sampleInputID, product: .productID(self.sampleProductID), quantity: 1)
+            self.createOrder(on: synchronizer, input: initialInput)
+
+            let input = OrderSyncProductInput(id: self.sampleInputID, product: .productID(self.sampleProductID), quantity: 2)
+            synchronizer.setProduct.send(input)
+        }
+
+        // Then
+        XCTAssertTrue(submittedItems.isNotEmpty)
+        for item in submittedItems {
+            XCTAssertTrue(item.total.isNotEmpty)
+            XCTAssertTrue(item.subtotal.isNotEmpty)
```

Comment:
> Can we set and check for specific values here? 

Definitely, changed here: https://github.com/woocommerce/woocommerce-ios/pull/9273/commits/b70a568fd73035f1c3f826a41195d90c2382cdd3 . 

> I don't think we need to provide totals to the API. Perhaps it's needed for our UI purposes?

We don't need to send these to the API indeed, from what I understand (apart for UI purposes), I believe these tests are there to assure that we still persist the original order data when we update it, since we pass this object through `ProductInputTransformer` and `OrderTotalsCalculator` internally and expect the Order to be updated with the combination of different OrderItems.

In this case the total and subtotal will be zero because we've used only the `productID` approach to create and update the OrderItem ( so only ID and quantity are needed), but I added additional tests [here](https://github.com/woocommerce/woocommerce-ios/blob/b70a568fd73035f1c3f826a41195d90c2382cdd3/WooCommerce/WooCommerceTests/ViewRelated/Orders/Order%20Creation/Synchronizer/RemoteOrderSynchronizerTests.swift#L564) to confirm that updating an OrderItem with other values (like having a previous price) keep them as well.